src/libraries/System.Text.Json/gen/Reflection/PropertyInfoWrapper.cs

```diff
@@ -5,18 +5,21 @@
 using System.Globalization;
 using System.Reflection;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 
 namespace System.Text.Json.Reflection
 {
     internal class PropertyInfoWrapper : PropertyInfo
     {
         private readonly IPropertySymbol _property;
-        private MetadataLoadContextInternal _metadataLoadContext;
+        private readonly MetadataLoadContextInternal _metadataLoadContext;
 
         public PropertyInfoWrapper(IPropertySymbol property, MetadataLoadContextInternal metadataLoadContext)
         {
             _property = property;
             _metadataLoadContext = metadataLoadContext;
+
+            NeedsAtSign = SyntaxFacts.GetKeywordKind(_property.Name) != SyntaxKind.None || SyntaxFacts.GetContextualKeywordKind(_property.Name) != SyntaxKind.None;
```

Comment:
could this live in `PropertyInfo`? Seems code is almost the same for FieldInfoWrapper and PropertyInfoWrapper

Reply:
@krwq - I didn't do this, but I noticed you've approved anyway. 

There's no common ancestry between `PropertyInfoWrapper`, `FieldInfoWrapper`, and `PropertyInfo`.

If it were to go in `PropertyInfo`, then it'd need to be implemented in:
![image](https://user-images.githubusercontent.com/263416/173246672-62219c56-6c32-48f3-8731-27a102804314.png)

Which would still mean it'd have to go in `FieldInfoWrapper.

But if it were to go in `MemberInfo`, then it would need implementing in lots more derived types:
![image](https://user-images.githubusercontent.com/263416/173246723-f80f0850-b685-472d-83f9-33cf753f19a1.png)

.. and it looks like, by adding it, that it might be the odd one out: none of those reference `Microsoft.CodeAnalysis.CSharp`.

Perhaps we can revisit this is a separate PR?

Reply:
sounds good